modin/pandas/indexing.py

```diff
@@ -653,7 +653,7 @@ def __getitem__(self, key):
         --------
         pandas.DataFrame.loc
         """
-        if self.df.empty:
+        if len(self.df.index) == 0 and self.df.empty:
```

Comment:
Why do you need these changes?

Reply:
If the frame has only index, the following lambda fails with KeyError.

Reply:
Is the problem specific to OmniSci backend or to all backends? Probably, the problem is in backend implementation of `empty`?

Reply:
Actually, I could find only one implementation of this method - https://github.com/modin-project/modin/blob/master/modin/pandas/dataframe.py#L325

Reply:
I don't understand why making a condition for default to pandas more strict might resolve any issue. We should be able to always be able to default to pandas.

Reply:
You are right. Actually, the problem is not in default_to_pandas() but in loc[], which is called from lambda. Pandas returns an empty Series for df.loc[1], but modin with omnisci backend fails with KeyError: 'Column F_1 does not exist in schema'. This issue should be fixed instead.